src/script/lib-build.sh

```diff
@@ -71,6 +71,15 @@ function discover_compiler() {
             break
         fi
     done
+    # but if this is {centos,rhel} we need gcc-toolset
+    if [ -f "/opt/rh/gcc-toolset-11/enable" ]; then
```

Comment:
now that GTS-12 is out on el9. and GCC-13 will be out early next year. i'd suggest use a loop for detecting GTS, so this script is more future proof, something like
```sh
for i in {13..11}; do
  # ...
done
```

Reply:
Unfortunately, I tried something like this and there's a version of gcc-toolset-12 present in the centos 8 container that fails the "configure" step. I would like to defer this until I can test on centos 9 stream (and this is blocked on a copr repository issue).

Reply:
this is interesting. i've been building Ceph on rocky linux 9 with GTS-12 for a months. what was the failure you ran into? but this is not a blocker. i am just curious.

Reply:
The failure was on centos (stream)8. I'm not sure why but it seems that there's a partial (?) install of 12 when 11 is installed. I plan on investigating it but for now only gcc-toolset-11 works on centos 8.